packages/url_launcher/url_launcher_platform_interface/test/method_channel_url_launcher_test.dart

```diff
@@ -41,6 +43,18 @@ void main() {
     final List<MethodCall> log = <MethodCall>[];
     channel.setMockMethodCallHandler((MethodCall methodCall) async {
       log.add(methodCall);
+
+      if (methodCall.method == 'canLaunch') {
+        if (methodCall.arguments['url'] == 'http://example.com/null') {
+          return null;
+        }
+      }
+
+      if (methodCall.method == 'launch') {
+        if (methodCall.arguments['url'] == 'http://example.com/null') {
+          return null;
+        }
+      }
     });
```

Comment:
nit: what about moving this code to the specific test where it's being used, or the `setUp` function in the worst case? This way the test doesn't leak state, and it doesn't need `log.clear();` in the `tearDown`.

Reply:
The problem is that setting the mock method channel (using `setMockMethodCallHandler`) has a global impact (this is how it is handled inside the `MethodChannel` class). So if I am to update in the specific test it will affect all tests that are run after (or during, since tests can run in parallel) the specific test. Or am I missing something here?

For now I removed the above logic, as it turns out the method channel will automatically return `null` if no return statement is specified, which is exactly the behavior needed by the tests. I did however added a `return null;` statement to make the logic explicit instead of implicitly relying on the behavior of the implementation of the `MethodChannel` class.